app/client/cli/debug.go

```diff
@@ -105,13 +105,17 @@ func NewDebugCommand() *cobra.Command {
 
 			setValueInCLIContext(cmd, busCLICtxKey, bus)
 
-			// TECHDEBT: simplify after P2P module consolidation.
-			var err error
-			p2pMod, err = getP2PModule(runtimeMgr)
+			mod, err := p2p.Create(bus)
 			if err != nil {
 				logger.Global.Fatal().Err(err).Msg("Failed to create p2p module")
 			}
 
+			var ok bool
+			p2pMod, ok = mod.(modules.P2PModule)
+			if !ok {
+				logger.Global.Fatal().Msgf("unexpected P2P module type: %T", mod)
+			}
+
```

Comment:
NB: This cannot be simplified without additional changes because `p2pMod` is a global var which is referenced in other functions in this module.